routes/v0/cases.js

```diff
@@ -1,38 +1,78 @@
-const { Router } = require('express');
 const models = require('../../models');
+const { SafeRouter } = require('../../permissions');
 
 const { Contact, Case, CaseAudit, sequelize } = models;
 const ContactController = require('../../controllers/contact-controller')(Contact);
 const CaseController = require('../../controllers/case-controller')(Case, sequelize);
 const CaseAuditController = require('../../controllers/case-audit-controller')(CaseAudit);
+const { can } = require('../../permissions');
+const { asyncHandler } = require('../../utils');
 
-const casesRouter = Router();
+/**
+ * This middleware checks if the user can edit the case.
+ * If yes, it sets the req.authorized to true.
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+const creatorCanEditCase = async (req, res, next) => {
+  if (!req.authorized) {
+    const { id } = req.params;
+    const caseObj = await CaseController.getCase(id);
+    const canEdit = can(req.user, 'edit', caseObj);
 
-casesRouter.get('/', async (req, res) => {
+    if (canEdit) {
+      req.authorized = true;
+    }
+  }
+
+  next();
+};
+
+/**
+ * A fake middleware that just marks the request as authorized.
+ * This can be deleted after we have all authorization middlewares set
+ * for all enpoints.
+ * @param {*} req
+ * @param {*} res
+ * @param {*} next
+ */
+const fakeAuthorized = (req, res, next) => {
```

Comment:
It would be fine to keep this and call it 'open' or 'openEndpoint', etc.  Some endpoints really are open to the world.  We just don't want that to be the default :)